File: benchmark/comparisons/script/scripts/fib_recursive.lua
-- fib_recursive.lua -- naive recursive Fibonacci, measures function-call overhead.

local function fib(n)
    if n < 2 then return n end
    return fib(n - 2) + fib(n - 1)
end

local count = tonumber(arg[1])

local t0 = os.clock()
local result = fib(count)
local elapsed_ns = (os.clock() - t0) * 1e9

print(math.floor(elapsed_ns))

File: benchmark/comparisons/script/scripts/fib_recursive.py
# fib_recursive.py — naive recursive Fibonacci, measures function-call overhead.
# argv[1] is the n in fib(n). fib(30) does ~1.6M calls.

import sys
import time


def fib(n):
    if n < 2:
        return n
    return fib(n - 2) + fib(n - 1)


count = int(sys.argv[1])

t0 = time.perf_counter_ns()
result = fib(count)
elapsed = time.perf_counter_ns() - t0

print(elapsed)
